File: src/main/java/com/silverhetch/clotho/connection/Get.java
package com.silverhetch.clotho.connection;

import java.net.HttpURLConnection;

/**
 * Created by Larry Hsiao on 2016/11/20.
 */

public class Get extends WebApi {

    public Get(Target target){
        super(target);
    }
    @Override
    protected void setupConnection(HttpURLConnection connection) throws Exception {
    }
}

File: src/main/java/com/silverhetch/clotho/connection/Get.kt
package com.silverhetch.elizabeth.arch.connection

import com.silverhetch.clotho.connection.Target
import com.silverhetch.clotho.connection.WebApi
import java.net.HttpURLConnection

/**
 * Http Get with Headers.
 */
class Get(target: Target, private val headers: Headers = EmptyHeaders()) : WebApi(target) {
    override fun setupConnection(connection: HttpURLConnection?) {
        headers.values().entries.forEach {
            connection!!.addRequestProperty(it.key, it.value)
        }
    }
}